components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/FirefoxAccount.kt

```diff
@@ -316,9 +317,40 @@ class FirefoxAccount(handle: FxaHandle, persistCallback: PersistCallback?) : Aut
         state: String,
         accessType: String = "online"
     ): String {
+        // This exists to allow clients that don't
+        // require keys to be able to call in without worrying about
+        // setting the params
         val scope = scopes.joinToString(" ")
+
+        val authParams = AuthorizationParams(
+                clientId,
+                scope,
+                state,
+                accessType,
+                null
+        )
+        return authorizeOAuthCode(authParams)
+    }
+
+    /**
+     * Provisions an OAuth code using the session token from state
+     * To be used directly by clients that require keys
+     *
+     * @param authParams Parameters needed for authorization request
+     * This performs network requests, and should not be used on the main thread.
+     */
+    fun authorizeOAuthCode(
```

Comment:
As discussed on Slack, since only a-c (and no downstream customer) uses this, I think it's worth breaking the API compatibility to keep only 1 `authorizeOAuthCode` method.